addon/components/o-s-s/private/base-modal.ts

```diff
@@ -7,9 +7,10 @@ export interface BaseModalArgs {
   close(): void;
 }
 
-export default class BaseModal extends Component<BaseModalArgs> {
+export default class BaseModal<T extends BaseModalArgs> extends Component<T> {
```

Comment:
This allows us to properly inherit from this component without having any ts-ignores